internal/databases/sqlserver/blob/blob.go

```diff
@@ -147,39 +147,9 @@ func (idx *Index) PutBlockList(xblocklist *XBlockListIn) ([]string, error) {
 		}
 		block.Offset = size
 		newBlocks = append(newBlocks, block)
-		switch xb.Mode {
-		case BlockCommitted:
-			if block.CommittedRev == 0 {
-				return nil, fmt.Errorf("proxy: blocklist operation block %s is not committed", xb.ID)
-			}
-			if block.UploadedRev != 0 {
-				block.UploadedRev = 0
-				block.UploadedSize = 0
-				garbage = append(garbage, fmt.Sprintf("%s.%d", xb.ID, block.UploadedRev))
-			}
-		case BlockUncommitted:
-			if block.UploadedRev == 0 {
-				return nil, fmt.Errorf("proxy: blocklist operation block %s is not committed", xb.ID)
-			}
-			if block.CommittedRev != 0 {
-				garbage = append(garbage, fmt.Sprintf("%s.%d", xb.ID, block.CommittedRev))
-			}
-			block.CommittedRev = block.UploadedRev
-			block.CommittedSize = block.UploadedSize
-			block.UploadedRev = 0
-			block.UploadedSize = 0
-		case BlockLatest:
-			if block.UploadedRev != 0 {
-				if block.CommittedRev != 0 {
-					garbage = append(garbage, fmt.Sprintf("%s.%d", xb.ID, block.CommittedRev))
-				}
-				block.CommittedRev = block.UploadedRev
-				block.CommittedSize = block.UploadedSize
-				block.UploadedRev = 0
-				block.UploadedSize = 0
-			}
-		default:
-			panic(fmt.Sprintf("unexpected block mode: %s", xb.Mode))
+		err := checkBlockMode(block, xb, &garbage)
```

Comment:
nope. The readability suffers.

Reply:
I removed the changes and added gocyclo ignore flag in 7437f84